frontend/src/metabase/visualizations/lib/renderer_utils.js

```diff
@@ -243,13 +243,15 @@ export function syntheticStackedBarsForWaterfallChart(datas) {
   const mainValues = mainSeries.map(d => d[1]);
   const totalValue = mainValues.reduce((total, value) => total + value, 0);
   const total = ["Total", totalValue];
-  // $FlowFixMe cloning for the total bar
-  total._origin = {
-    seriesIndex: mainSeries[0]._origin.seriesIndex,
-    rowIndex: mainSeries.length,
-    cols: mainSeries[0]._origin.cols,
-    row: total,
-  };
+  if (mainSeries[0]._origin) {
```

Comment:
This is just an extra safeguard discovered through the tests (cause the tests did not create a full-blown series object).